sunpy/util/sysinfo.py

```diff
@@ -2,14 +2,42 @@
 This module provides functions to retrieve system information.
 """
 import platform
+from collections import defaultdict
+from importlib.metadata import PackageNotFoundError, version, requires, distribution
 
-from pkg_resources import get_distribution
+from packaging.requirements import Requirement
 
 import sunpy.extern.distro as distro
 
 __all__ = ['system_info', 'find_dependencies', 'missing_dependencies_by_extra']
 
 
+def get_requirements(package):
+    """
+    This wraps `importlib.metadata.requires` to not be garbage.
+
+    Parameters
+    ----------
+    package : str
+        Package you want requirements for.
+
+    Returns
+    -------
+    `dict`
+        A dictionary of requirements with keys being the extra requirement group names.
+    """
+    requirements: list = requires(package)
+    requires_dict = defaultdict(list)
+    for requirement in requirements:
+        package = Requirement(requirement).name
+        if ";" in requirement:
```

Comment:
you can use `Requirement().marker` to extract what's after the `;` which at least marginally cleans this up (as you can do `if req.marker` rather than `if ';' in`.

This is still upsetting though.

Reply:
Its all upsetting.